File: Code Samples/CMSC 12/1. Loops or Iterations/3. loop versions/3.2. while-loop/while-loop.py
''' This code is a while-loop version of the original version from item 3.0. '''

# variable declarations
num_variable = 0

# STOP if already in this value!
exit_value = 10


# while-loop version 
while (num_variable <= 10):    # <- do until variable <= 10!
    num_variable += 1          # <- update the number
    print(num_variable)        # <- print the number



# further explanation below for your reference

'''
#====== while-loop version (1) ======#      #========= linear version (1) =========#

num_variable = 0                            num_variable = 0

while (num_variable <= 1):                  # iteration 1
    num_variable += 1                       num_variable += 1
    print(num_variable)                     print(num_variable)     <- prints what?
    
                                            # check if should stop now
                                            if (num_variable <= 3):
                                                exit(0)                                                

#==================================#        #======================================#








#====== while-loop version (3) ======#      #========= linear version (3) =========#

num_variable = 0                            num_variable = 0

while (num_variable <= 3):                  # iteration 1
    num_variable += 1                       num_variable += 1
    print(num_variable)                     print(num_variable)     <- prints what?
    
                                            # check if should stop now
                                            if (num_variable <= 3):
                                                exit(0)
                                                
                                                
                                                
                                            # iteration 2
                                            num_variable += 1
                                            print(num_variable)     <- prints what?
    
                                            # check if should stop now
                                            if (num_variable <= 3):
                                                exit(0)
                                                
                                                
    
                                            # iteration 3
                                            num_variable += 1
                                            print(num_variable)     <- prints what?
    
                                            # check if should stop now
                                            if (num_variable <= 3):
                                                exit(0)
                                                
                                                
    
                                            # iteration 4
                                            num_variable += 1
                                            print(num_variable)     <- prints what?
    
                                            # check if should stop now
                                            if (num_variable <= 3):
                                                exit(0)
                                                
                                            # ... .. ... until infinity ... .. ... #
    
#==================================#        #======================================#







#====== while-loop version (10) =====#      #========= linear version (10) ========#

num_variable = 0                            num_variable = 0

while (num_variable <= 10):                 # iteration 1
    num_variable += 1                       num_variable += 1
    print(num_variable)                     print(num_variable)     <- prints what?
    
                                            # check if should stop now
                                            if (num_variable <= 3):
                                                exit(0)
                                                

    
                                            # iteration 2
                                            num_variable += 1
                                            print(num_variable)     <- prints what?
    
                                            # check if should stop now
                                            if (num_variable <= 3):
                                                exit(0)
                                                

    
                                            # iteration 3
                                            num_variable += 1
                                            print(num_variable)     <- prints what?
    
                                            # check if should stop now
                                            if (num_variable <= 3):
                                                exit(0)
                                                

        
                                            # iteration 4
                                            num_variable += 1
                                            print(num_variable)     <- prints what?
    
                                            # check if should stop now
                                            if (num_variable <= 3):
                                                exit(0)
                                                

    
                                            # iteration 5
                                            num_variable += 1
                                            print(num_variable)     <- prints what?
    
                                            # check if should stop now
                                            if (num_variable <= 3):
                                                exit(0)
                                                

        
                                            # iteration 6
                                            num_variable += 1
                                            print(num_variable)     <- prints what?
    
                                            # check if should stop now
                                            if (num_variable <= 3):
                                                exit(0)
                                                

    
                                            # iteration 7
                                            num_variable += 1
                                            print(num_variable)     <- prints what?
    
                                            # check if should stop now
                                            if (num_variable <= 3):
                                                exit(0)
                                                

        
                                            # iteration 8
                                            num_variable += 1
                                            print(num_variable)     <- prints what?
    
                                            # check if should stop now
                                            if (num_variable <= 3):
                                                exit(0)
                                                

    
                                            # iteration 9
                                            num_variable += 1
                                            print(num_variable)     <- prints what?
    
                                            # check if should stop now
                                            if (num_variable <= 3):
                                                exit(0)
                                                

        
                                            # iteration 10
                                            num_variable += 1
                                            print(num_variable)     <- prints what?
    
                                            # check if should stop now
                                            if (num_variable <= 3):
                                                exit(0)
                                                

        
                                            # iteration 11
                                            num_variable += 1
                                            print(num_variable)     <- prints what?
    
                                            # check if should stop now
                                            if (num_variable <= 3):
                                                exit(0)
                                                

        
                                            # iteration 12
                                            num_variable += 1
                                            print(num_variable)     <- prints what?
    
                                            # check if should stop now
                                            if (num_variable <= 3):
                                                exit(0)
                                                

                                            # ... .. ... until infinity ... .. ... #
    
#==================================#        #======================================#


'''

File: CMSC 21/1. Loops or Iterations/4. while loop/while-loop.c
#include <stdio.h>

/* This code is a while-loop version of the original version from item #2. */

// While-loop version
int main() {

    // variable declarations
    int num_variable = 0;
    int current_number = 0; 

    // Start of while loop 
    while (current_number < 10) {
        num_variable += 1;            // Update the number
        printf("%d\n", num_variable); // Print the number
        current_number++;             // Update the loop control variable
    }

    // End of code
    return 0;
}



/* further explanation below for your reference


#====== while-loop version (1) ======#        #========= linear version (1) =========#

int num_variable = 0;                       int num_variable = 0; 
int current_number = 0;                     int current_number = 0; 
                                            const int exit_value = 1;


while (current_number < 10) {               // Iteration 1
    num_variable++;                         num_variable += 1;
    printf("%d\n", num_variable);           printf("%d\n", num_variable);
    current_number++;                       current_number++;
}
                                            // Check if done?
                                            if (num_variable == exit_value) {
                                                exit(0);
                                            }
                                            
#==================================#        #======================================#





#====== while-loop version (3) ======#        #========= linear version (3) =========#

int num_variable = 0;                       int num_variable = 0; 
int current_number = 0;                     int current_number = 0; 
                                            const int exit_value = 3;


while (current_number < 10) {               // Iteration 1
    num_variable++;                         num_variable += 1;
    printf("%d\n", num_variable);           printf("%d\n", num_variable);
    current_number++;                       current_number++;
}
                                            // Check if done?
                                            if (num_variable == exit_value) {
                                                exit(0);
                                            }


                                            // Iteration 2 
                                            num_variable += 1;
                                            printf("%d\n", num_variable);
                                            current_number++;

                                            // Check if done?
                                            if (num_variable == exit_value) {
                                                exit(0);
                                            }    


                                            // Iteration 3
                                            num_variable += 1;
                                            printf("%d\n", num_variable);
                                            current_number++;

                                            // Check if done?
                                            if (num_variable == exit_value) {
                                                exit(0);
                                            }
    

#==================================#        #======================================#





#====== while-loop version (10) =====#        #========= linear version (10) ========#

int num_variable = 0;                       int num_variable = 0; 
int current_number = 0;                     int current_number = 0; 
                                            const int exit_value = 10;


while (current_number < 10) {               // Iteration 1
    num_variable++;                         num_variable += 1;
    printf("%d\n", num_variable);           printf("%d\n", num_variable);
    current_number++;                       current_number++;
}
                                            // Check if done?
                                            if (num_variable == exit_value) {
                                                exit(0);
                                            }


                                            // Iteration 2 
                                            num_variable += 1;
                                            printf("%d\n", num_variable);
                                            current_number++;

                                            // Check if done?
                                            if (num_variable == exit_value) {
                                                exit(0);
                                            }    


                                            // Iteration 3
                                            num_variable += 1;
                                            printf("%d\n", num_variable);
                                            current_number++;

                                            // Check if done?
                                            if (num_variable == exit_value) {
                                                exit(0);
                                            }


                                            // Iteration 4
                                            num_variable += 1;
                                            printf("%d\n", num_variable);
                                            current_number++;

                                            // Check if done?
                                            if (num_variable == exit_value) {
                                                exit(0);
                                            }


                                            // Iteration 5
                                            num_variable += 1;
                                            printf("%d\n", num_variable);
                                            current_number++;

                                            // Check if done?
                                            if (num_variable == exit_value) {
                                                exit(0);
                                            }


                                            // Iteration 6
                                            num_variable += 1;
                                            printf("%d\n", num_variable);
                                            current_number++;

                                            // Check if done?
                                            if (num_variable == exit_value) {
                                                exit(0);
                                            }


                                            // Iteration 7
                                            num_variable += 1;
                                            printf("%d\n", num_variable);
                                            current_number++;

                                            // Check if done?
                                            if (num_variable == exit_value) {
                                                exit(0);
                                            }


                                            // Iteration 8
                                            num_variable += 1;
                                            printf("%d\n", num_variable);
                                            current_number++;

                                            // Check if done?
                                            if (num_variable == exit_value) {
                                                exit(0);
                                            }


                                            // Iteration 9
                                            num_variable += 1;
                                            printf("%d\n", num_variable);
                                            current_number++;

                                            // Check if done?
                                            if (num_variable == exit_value) {
                                                exit(0);
                                            }


                                            // Iteration 10
                                            num_variable += 1;
                                            printf("%d\n", num_variable);
                                            current_number++;

                                            // Check if done?
                                            if (num_variable == exit_value) {
                                                exit(0);
                                            }
    
#==================================#        #======================================#


'''

*/